behave/features/steps/cla_pick_up_digital_case_by_call_back.py

```diff
@@ -0,0 +1,96 @@
+from cla_common.call_centre_availability import OpeningHours
+from cla_common.constants import OPERATOR_HOURS
+from features.constants import CLA_CALLBACK_CASES, CLA_FRONTEND_URL
+from features.steps.cla_in_scope import assert_header_on_page
+import random
+
+
+@given(u'that I have created cases with callbacks')
+# This is the background step that takes test cases and assigns a callback slot some time over the next few days
+def step_impl(context):
+    # format of the json that we need for the callback is
+    # {
+    #     "datetime": "05/05/2022 09:00",
+    #     "notes": "",
+    #     "priority_callback": false
+    # }
+    # Need to get available slots from cla_common.
+    # Use OpeningHours class with correct setup so that it knows about bank holidays
+    # Choose 6 days so can be sure that will not get three bank holidays/sunday etc
+    operator_hours = OpeningHours(**OPERATOR_HOURS)
+    available_days_from_common = operator_hours.available_days(6)
+    all_available_slots = []
+    for day in available_days_from_common:
+        available_slots = operator_hours.time_slots(day.date())
+        all_available_slots.extend(available_slots)
+    # do every other slot as on the display they lump two slots together on website
+    slots_chosen = all_available_slots[:8:2]
+    slots_chosen.append(all_available_slots[0])
```

Comment:
What do you think of putting all these in another function as it's quite complex, I think maybe moving it to another place would make the current function easier to read as you have to keep less details in your head when reading it

Reply:
will do